### ISSUE TITLE
App turns black if simultaneously showing alert to allow PushNotifications and alert from Siren

### ISSUE THREAD

**Issue by iKefir** [ArtSabintsev/Siren#341]
## If you're experiencing a problem integrating Siren into your app, please provide the following information when posting a new issue:

- **Are you using the [latest version](https://github.com/ArtSabintsev/Siren/releases) of Siren?**: Yes
- **What is your app's Bundle ID?**: like in Example app
- **When was the latest version of your app published to the App Store?**: 
- **Is your app published in the US App Store? If not, what App Store is it published in?**:
- **Does Siren work if you plugin your app's `BundleID` (and `countryCode`, if necessary) into the Example app?**: still have an issue

I saw similar topics in https://github.com/ArtSabintsev/Siren/issues/338 and https://github.com/ArtSabintsev/Siren/issues/335, however I have some additional information to share.

I use Xcode 11.5 and latest version of Siren (5.4.4) installed using CocoaPods. 

I tested my sample app on iOS 12.4, iOS 13.3 and iOS 13.5. On iOS 12 everything worked like it should, but on both iOS 13 versions I see black screen instead of app content.

Additionally I figured out that issue is not just with PushNotifications alert. If I ask for permission to use camera simultaneously (for some reason) with Siren call - I also get black screen. Probably the same behaviour can happen with other calls for permission. If I make simultaneous calls for two different permissions (PushNotifications and camera permissions) everything works fine, but when I additionally make Siren call - screen becomes black.

Foreseeing your question about whether I tested it in your sample app. Yes, I did, but it does not work properly for me on iOS 13 even if I don't use Siren. On iOS 12 I see content of an app behind alert, but on iOS 13 I see black screen.

You can download my sample app [here](https://drive.google.com/drive/folders/19kQPSyBApNz3SFy7-5KxaLQvPwFDNqt4?usp=sharing) as well as see videos on which I recorded described behaviour.

**Comment by ArtSabintsev** [comment]
Awesome - thanks for all the info! I will take a look.

**Comment by phillipjacobs** [comment]
**So I just experienced something similar. I'm also on iOS 13 using the latest Siren.** 

My screen didn't turn black, the permission alerts just glitched for a second and after answering all my permission alerts, Siren just never popped up. After running the app (code unchanged) again, the issue of course didn't happen again as there were no permission alerts.

What worked for me is always calling siren first.

```
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        window?.makeKeyAndVisible()
        Siren.shared.wail()
       // check whatever permissions here
}
```

With all that said, @iKefir i'd say using @ArtSabintsev's quick fix [here](https://github.com/ArtSabintsev/Siren/issues/338#issuecomment-643667060) is not just a quick fix but also best practice by far. Running tasks like those simultaneously is bound to cause issues.

**Comment by ArtSabintsev** [comment]
@phillipjacobs Thanks for following up! 

When I worked at The Washington Post, what we did was create a permission hierarchy internally. So we would first check to see if any other alert was being presented, and if the system didn't show another alert, we'd present Siren (if we had a need to do so).

I think this is the only appropriate solution here. 

@phillipjacobs's solution works as well if you want to take easier / less-code-intensive route (cc @iKefir).

Closing this issue, but feel free to follow up!

**Comment by iKefir** [comment]
Solution from @phillipjacobs didn't work for me, but what did work was changing code in `PresentationManager` for function `createWindow`

from 

```
private func createWindow() -> UIWindow {
    var window = UIWindow()
    if #available(iOS 13.0, *) {
        guard let windowScene = UIApplication.shared.connectedScenes
            .first(where: { $0.activationState == .foregroundActive }) as? UIWindowScene else { return UIWindow() }
        window = UIWindow(windowScene: windowScene)
    } else {
        window = UIWindow(frame: UIScreen.main.bounds)
    }

    window.windowLevel = UIWindow.Level.alert + 1

    let viewController = SirenViewController()
    viewController.retainedWindow = window

    window.rootViewController = viewController
    return window
}
```

to

```
private func createWindow() -> UIWindow {
    var window = UIWindow()
    if #available(iOS 13.0, *),
        let windowScene = UIApplication.shared.connectedScenes
            .first(where: { $0.activationState == .foregroundActive }) as? UIWindowScene
    {
        window = UIWindow(windowScene: windowScene)
    } else {
        window = UIWindow(frame: UIScreen.main.bounds)
    }

    window.windowLevel = UIWindow.Level.alert + 1

    let viewController = SirenViewController()
    viewController.retainedWindow = window

    window.rootViewController = viewController
    return window
}
```

The problem seems to be that if system notification popup is already on screen - guard expression returns UIWindow() and that window is what I was seeing as black screen. I admit that problem could be with my app, because I intentionally created it without SceneDelegate, but every old app can work fine on newest iOS without configuring SceneDelegate.

I don't know If my solution should be incorporated in Siren, but would propose to at least add section in README in which this issue will be described so that new users of Siren would pay attention to whether it can happen in their app.
